### ISSUE TITLE
Stargate is not able to join DSE cluster after a quick restart

### ISSUE THREAD

**Issue by dimas-b** [stargate/stargate#587]
When a Stargate node connected to DSE is stopped and quickly restarted (the pause must be less than 30 sec), it is not able to reconnect and the cluster state looks like:

```
$ ./nodetool --host 127.0.0.1 describecluster
Cluster Information:
	Name: dse1
	Snitch: com.datastax.bdp.snitch.DseSimpleSnitch
	DynamicEndPointSnitch: enabled
	Partitioner: org.apache.cassandra.dht.Murmur3Partitioner
	Schema versions:
		caf667b8-2c33-36f6-a765-1b55a1a14a01: [127.0.0.1]
		UNREACHABLE: [127.0.2.1]
```

(`127.0.2.1` is the IP of the Stargate node).

Side note: Running Stargate with `-Dcassandra.allow_unsafe_join=true` (this option is not set by default) allows it to connect even with a short down time.

Note: If Stargate is down for more than 1 min, DSE will discard its internal state about the Stargate node (log messages below) and Stargate will be able to reconnect when restarted.

```
INFO  [GossipTasks:1] 2021-01-11 16:20:22,824  Gossiper.java:1301 - InetAddress /127.0.2.1 is now DOWN
INFO  [GossipTasks:1] 2021-01-11 16:20:33,832  Gossiper.java:1043 - FatClient /127.0.2.1 has been silent for 30000ms, removing from gossip
INFO  [NonPeriodicTasks:1] 2021-01-11 16:20:33,834  CassandraDaemon.java:783 - All nodes in this cluster are on DSE version 6.8.8
```

**Comment by rchernobelskiy** [comment]
I ran into this one too btw

**Comment by marksurnin** [comment]
@dimas-b I came across this issue as well and it seems to happen when a Stargate node tries to join the cluster with the same IP as the node that has just been terminated.

As you pointed out, the old node is removed from gossip only after 30 seconds:
```
Gossiper.java:1043 - FatClient /127.0.2.1 has been silent for 30000ms, removing from gossip
```

I noticed this in the log right before Stargate shuts down. [Here](https://github.com/apache/cassandra/blob/cassandra-3.11.6/src/java/org/apache/cassandra/gms/Gossiper.java#L1655) is this line in the source code. It would have announced shutdown in the first branch of the if-statement.
```
Gossiper.java:1655 - No local state, state is in silent shutdown, or node hasn't joined, not announcing shutdown
```

What about explicitly announcing shutdown to other nodes via gossip? This would allow a new node to be launched with the same IP.
